MetadataExtractor/StringValue.cs

```diff
@@ -0,0 +1,172 @@
+            get { return (_buffer != null) ? _buffer.Length : 0; }
+        }
+
+        #region Conversion methods
+
+        public double ToDouble()
+        {
+            double output;
+            if (!double.TryParse(ToString(), out output))
+                throw new FormatException();
+
+            return output;
+        }
+
+        public float ToSingle()
+        {
+            return (float)ToDouble();
+        }
+
+        public string ToString(IFormatProvider provider) => ToString();
+
+        #region IConvertible
+        double IConvertible.ToDouble(IFormatProvider provider) => ToDouble();
+
+        decimal IConvertible.ToDecimal(IFormatProvider prodiver) => (decimal)ToDouble();
+
+        float IConvertible.ToSingle(IFormatProvider provider) => ToSingle();
+
+        TypeCode IConvertible.GetTypeCode() => TypeCode.Object;
+
+        bool IConvertible.ToBoolean(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        byte IConvertible.ToByte(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        char IConvertible.ToChar(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        DateTime IConvertible.ToDateTime(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        short IConvertible.ToInt16(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        int IConvertible.ToInt32(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        long IConvertible.ToInt64(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        sbyte IConvertible.ToSByte(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        ushort IConvertible.ToUInt16(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        uint IConvertible.ToUInt32(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        ulong IConvertible.ToUInt64(IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        object IConvertible.ToType(Type conversionType, IFormatProvider provider)
+        {
+            throw new NotImplementedException();
+        }
+
+        #endregion
+
+        #endregion
+
+        #region Formatting
+
+        public override string ToString()
+        {
+            return ToString(_encoder);
+        }
+        public string ToString(Encoding encoder)
+        {
+            return encoder.GetString(_buffer, 0, Length);
+        }
+
+        #endregion
+
+        // user-defined conversion from StringValue to Double
+        /*public static implicit operator double(StringValue strval)
+        {
```

Comment:
Yep. I tested them out, but came to a similar conclusion. It seemed a shame to just delete them, but removing is also fine.